### ISSUE TITLE
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /app/tools/copyAssets.ts

### ISSUE THREAD

**Issue by Galaxy83** [5e-bits/5e-srd-api#375]
docker compose up --build
[+] Building 17.2s (10/10) FINISHED                                                                                                                                                    docker:desktop-linux
 => [api internal] load .dockerignore                                                                                                                                                                  0.0s
 => => transferring context: 97B                                                                                                                                                                       0.0s
 => [api internal] load build definition from Dockerfile                                                                                                                                               0.0s
 => => transferring dockerfile: 297B                                                                                                                                                                   0.0s
 => [api internal] load metadata for docker.io/library/node:18-alpine                                                                                                                                  0.7s
 => [api 1/6] FROM docker.io/library/node:18-alpine@sha256:b1a0356f7d6b86c958a06949d3db3f7fb27f95f627aa6157cb98bc65c801efa2                                                                            0.0s
 => [api internal] load build context                                                                                                                                                                  0.1s
 => => transferring context: 40.64kB                                                                                                                                                                   0.0s
 => CACHED [api 2/6] WORKDIR /app                                                                                                                                                                      0.0s
 => CACHED [api 3/6] COPY package.json package-lock.json /app/                                                                                                                                         0.0s
 => CACHED [api 4/6] RUN npm install                                                                                                                                                                   0.0s
 => CACHED [api 5/6] COPY . /app                                                                                                                                                                       0.0s
 => ERROR [api 6/6] RUN npm run build                                                                                                                                                                 16.4s
------
 > [api 6/6] RUN npm run build:
0.571
0.571 > dnd-5e-srd-api@0.0.0 build
0.571 > npm-run-all clean bundle-swagger build:ts copy-assets
0.571
1.236
1.236 > dnd-5e-srd-api@0.0.0 clean
1.236 > rimraf dist/*
1.236
1.790
1.790 > dnd-5e-srd-api@0.0.0 bundle-swagger
1.790 > npm run validate-swagger && swagger-cli bundle --outfile src/swagger/dist/openapi.yml --type yaml src/swagger/swagger.yml && swagger-cli bundle --outfile src/swagger/dist/openapi.json --type json src/swagger/swagger.yml
1.790
2.214
2.214 > dnd-5e-srd-api@0.0.0 validate-swagger
2.214 > swagger-cli validate src/swagger/swagger.yml
2.214
2.985 src/swagger/swagger.yml is valid
3.561 Created src/swagger/dist/openapi.yml from src/swagger/swagger.yml
4.049 Created src/swagger/dist/openapi.json from src/swagger/swagger.yml
4.474
4.474 > dnd-5e-srd-api@0.0.0 build:ts
4.474 > tsc
4.474
15.46
15.46 > dnd-5e-srd-api@0.0.0 copy-assets
15.46 > ts-node tools/copyAssets.ts
15.46
16.32 TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /app/tools/copyAssets.ts
16.32     at new NodeError (node:internal/errors:405:5)
16.32     at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:136:11)
16.32     at defaultGetFormat (node:internal/modules/esm/get_format:182:36)
16.32     at defaultLoad (node:internal/modules/esm/load:101:20)
16.32     at nextLoad (node:internal/modules/esm/hooks:864:28)
16.32     at load (/app/node_modules/ts-node/dist/child/child-loader.js:19:122)
16.32     at nextLoad (node:internal/modules/esm/hooks:864:28)
16.32     at Hooks.load (node:internal/modules/esm/hooks:447:26)
16.32     at MessagePort.handleMessage (node:internal/modules/esm/worker:196:24)
16.32     at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20) {
16.32   code: 'ERR_UNKNOWN_FILE_EXTENSION'
16.32 }
16.35 ERROR: "copy-assets" exited with 1.

**Comment by tarasis** [comment]
Just hit this myself, did you manage to solve it? 

(I was building the docker image on Ubuntu 23.10 GNU/Linux 6.5.0-1013-oracle aarch64) (one of their free VPS's)

```
13.99 TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /app/tools/copyAssets.ts
13.99     at new NodeError (node:internal/errors:405:5)
13.99     at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:136:11)
13.99     at defaultGetFormat (node:internal/modules/esm/get_format:182:36)
13.99     at defaultLoad (node:internal/modules/esm/load:101:20)
13.99     at nextLoad (node:internal/modules/esm/hooks:864:28)
13.99     at load (/app/node_modules/ts-node/dist/child/child-loader.js:19:122)
13.99     at nextLoad (node:internal/modules/esm/hooks:864:28)
13.99     at Hooks.load (node:internal/modules/esm/hooks:447:26)
13.99     at MessagePort.handleMessage (node:internal/modules/esm/worker:196:24)
13.99     at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20) {
13.99   code: 'ERR_UNKNOWN_FILE_EXTENSION'
13.99 }
14.03 ERROR: "copy-assets" exited with 1.

```

**Comment by tarasis** [comment]
Just to update, have confirmed that issue also occurs on an amd64/x64_64 system

```
4.959 
4.959 > dnd-5e-srd-api@0.0.0 build:ts
4.959 > tsc
4.959 
15.16 
15.16 > dnd-5e-srd-api@0.0.0 copy-assets
15.16 > ts-node tools/copyAssets.ts
15.16 
16.22 TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /app/tools/copyAssets.ts
16.22     at new NodeError (node:internal/errors:405:5)
16.22     at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:136:11)
16.22     at defaultGetFormat (node:internal/modules/esm/get_format:182:36)
16.22     at defaultLoad (node:internal/modules/esm/load:101:20)
16.22     at nextLoad (node:internal/modules/esm/hooks:864:28)
16.22     at load (/app/node_modules/ts-node/dist/child/child-loader.js:19:122)
16.22     at nextLoad (node:internal/modules/esm/hooks:864:28)
16.22     at Hooks.load (node:internal/modules/esm/hooks:447:26)
16.22     at MessagePort.handleMessage (node:internal/modules/esm/worker:196:24)
16.22     at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20) {
16.22   code: 'ERR_UNKNOWN_FILE_EXTENSION'
16.22 }
16.25 ERROR: "copy-assets" exited with 1.
------
failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1
root@NanoKVM:/srv/dndapi/5e-srd-api# arch
x86_64
```

**Comment by Galaxy83** [comment]
> Just hit this myself, did you manage to solve it?

Nope

**Comment by bagelbits** [comment]
I have a solution. I'm going to merge it in shortly. There's an issue with `ts-node`:

https://github.com/TypeStrong/ts-node/issues/1997

**Comment by tarasis** [comment]
Thank you ❤️ , I haven’t touched TypeScript yet so couldn’t debug!On 7 Feb 2024, at 20:10, github-actions[bot] ***@***.***> wrote:﻿
🎉 This issue has been resolved in version 3.4.1 🎉
The release is available on GitHub release
Your semantic-release bot 📦🚀
